remo/profiles/views.py

```diff
@@ -195,10 +195,10 @@ def view_profile(request, display_name):
             messages.info(request, mark_safe(msg))
 
     if nominee_form.is_valid():
-        if ((is_nomination_period or
-             waffle.switch_is_active('enable_rotm_tasks')) and
-                request.user.groups.filter(name='Mentor').exists()):
-            nominee_form.save()
+        if ((is_nomination_period or waffle.switch_is_active('enable_rotm_tasks')) and
+                request.user.groups.filter(name='Mentor').exists() and
+                request.user != user):
+            nominee_form.save(nominated_by=request.user)
```

Comment:
If you follow this paradigm, then you don't need to make any changes in the `save` method.